source/extensions/filters/http/basic_auth/basic_auth_filter.cc

```diff
@@ -31,18 +32,16 @@ FilterConfig::FilterConfig(UserMap&& users, const std::string& forward_username_
     : users_(std::move(users)), forward_username_header_(forward_username_header),
       stats_(generateStats(stats_prefix + "basic_auth.", scope)) {}
 
-bool FilterConfig::validateUser(absl::string_view username, absl::string_view password) const {
-  auto user = users_.find(username);
-  if (user == users_.end()) {
-    return false;
-  }
-
-  return computeSHA1(password) == user->second.hash;
-}
-
 BasicAuthFilter::BasicAuthFilter(FilterConfigConstSharedPtr config) : config_(std::move(config)) {}
 
 Http::FilterHeadersStatus BasicAuthFilter::decodeHeaders(Http::RequestHeaderMap& headers, bool) {
+  const auto* route_specific_settings =
+      Http::Utility::resolveMostSpecificPerFilterConfig<FilterConfigPerRoute>(decoder_callbacks_);
+  const UserMap* users = &config_->users();
+  if (route_specific_settings != nullptr) {
+    users = &route_specific_settings->users();
```

Comment:
nit: Prefer reference_wrapper because it never be null. But pointer is also OK if you prefer it.

Reply:
I do not prefer pointers. I used the pointer to reassign `users` inside the if statement, and I couldn't do that with `reference_wrapper`.

Reply:
I think `reference_wraper` also could do the reassign? But anyway, it's fine.